docs/kthena/blog/modelserving/index.md

```diff
@@ -0,0 +1,185 @@
+---
+slug: modelserving-blog-post
+title: A Deep Dive of the Kthena's ModelServing
+authors: [LiZhenCheng9527]
+tags: []
+---
+
+# A Deep Dive of the Kthena's ModelServing
+
+## Introduction
+
+As large models continue to grow exponentially in parameter size, the resource limits of a single virtual or physical machine can no longer meet their demands. To address this challenge, the industry has introduced innovative strategies such as PD-diaggregation deployment and hybrid deployment of large and small models. These approaches have transformed inference execution: instead of a single Pod handling an entire inference task, multiple Pods now often collaborate to complete a single prediction. This multi-Pod collaboration has become a key trend in large model inference deployment.
+
+In practice, inference models may still run within a single Pod (as in traditional single-node scenarios), across a group of identical Pods (for larger models), or among Pods with specialized roles (as in PD-disaggregation deployments). This flexible deployment not only improves resource utilization but also enables more efficient large model inference.
+
+`ModelServing` is a specialized component of `Kthena` designed to manage and orchestrate the lifecycle of inference model workloads. It can conveniently represent and manage multiple deployment models, such as `PD-disaggregation`, `tensor parallelism`, `pipeline parallelism`, and single-GPU model deployment, because its three-tier architecture.
+
+<!-- truncate -->
+
+## Three-tier Architecture
+
+`ModelServing` adopts a three-tier architecture `ModelServing → servinggroups → roles` to address the limitations of Kubernetes' traditional two-tier architecture (e.g., deployment and statefulSet) in managing diverse inference workload deployment scenarios. The architecture diagram is shown below:
+
+![ModelServing Architecture](images/modelServing_architecture.svg)
+
+- **ModelServing:** The central component responsible for managing the lifecycle of inference model workloads. It offers a unified interface for deploying, monitoring, and managing inference models, as well as querying their statuses.
+- **ServingGroups:** An `ServingGroup` is a collection of Roles, each fulfilling a specific function within the inference process, such as `Prefill` or `Decode`.
```

Comment:
This desc seems not right, each group can fullfill the whole inference, it include both Prefill and Decode roles for PD-Disaggregation deployment.